crates/revm/src/builder.rs

```diff
@@ -1,21 +1,22 @@
 use crate::{
     db::{Database, DatabaseRef, EmptyDB, WrapDatabaseRef},
-    handler::register,
-    primitives::{
-        BlockEnv, CfgEnv, CfgEnvWithHandlerCfg, Env, EnvWithHandlerCfg, HandlerCfg, SpecId, TxEnv,
+    handler::{
+        register::{self, EvmHandler},
+        CfgEnvWithChainSpec, EnvWithChainSpec,
     },
-    Context, ContextWithHandlerCfg, Evm, Handler,
+    primitives::{CfgEnv, ChainSpec, Env, EthChainSpec, InvalidTransaction, TransactionValidation},
+    Context, ContextWithChainSpec, Evm, EvmContext, Handler,
 };
 use core::marker::PhantomData;
 use std::boxed::Box;
 
 /// Evm Builder allows building or modifying EVM.
 /// Note that some of the methods that changes underlying structures
 /// will reset the registered handler to default mainnet.
-pub struct EvmBuilder<'a, BuilderStage, EXT, DB: Database> {
-    context: Context<EXT, DB>,
+pub struct EvmBuilder<'a, BuilderStage, ChainSpecT: ChainSpec, EXT, DB: Database> {
```

Comment:
Instead of requiring `Default` for `ChainSpec::Transaction` and `ChainSpec::Block`, I think a nice design would be to have `EvmBuilder` contain `transaction: Option<ChainSpecT::Transaction>` and `block: Option<ChainSpecT::Block>`.

I feel that both should explicitly be set and when finishing the builder, a `Result` could be returned that constructs the `Context` or returns an error mentioning that no transaction and/or block was provided.

That way the user is not quietly using a default transaction or block configuration.

WDYT?